flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/table/FileSystemTableSource.java

```diff
@@ -272,7 +272,14 @@ public ScanRuntimeProvider getScanRuntimeProvider(ScanContext scanContext) {
     }
 
     private SourceProvider createSourceProvider(BulkFormat<RowData, FileSourceSplit> bulkFormat) {
-        return SourceProvider.of(FileSource.forBulkFileFormat(bulkFormat, paths()).build());
+        final FileSource.FileSourceBuilder<RowData> fileSourceBuilder =
+                FileSource.forBulkFileFormat(bulkFormat, paths());
+
+        tableOptions
+                .getOptional(FileSystemConnectorOptions.SOURCE_MONITOR_INTERVAL)
+                .ifPresent(fileSourceBuilder::monitorContinuously);
```

Comment:
btw what happens if the duration is set but 0?

Reply:
Added a doc